tests/karma/disco/components/TestAddon.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { renderIntoDocument } from 'react-addons-test-utils';
+import { findDOMNode } from 'react-dom';
+import Addon from 'disco/components/Addon';
+
+describe('<Addon type="Extension" />', () => {
+  let root;
+  const result = {
+    id: 'test-id',
+    type: 'Extension',
+    heading: 'test-heading',
+    subHeading: 'test-sub-heading',
+    editorialDescription: 'test-editorial-description',
+  };
+
+  beforeEach(() => {
+    root = renderIntoDocument(<Addon {...result} />);
+  });
+
+  it('renders the heading', () => {
+    assert.include(root.refs.heading.textContent, 'test-heading');
+  });
+
+  it('renders the editorial description', () => {
+    assert.equal(root.refs['editorial-description'].textContent, 'test-editorial-description');
+  });
+
+  it('renders the sub-heading', () => {
+    assert.equal(root.refs['sub-heading'].textContent, 'test-sub-heading');
+  });
+
+  it("doesn't render the subheading when not present", () => {
+    const data = Object.assign({}, result, { subHeading: undefined });
```

Comment:
Using the spread operator is a lot cleaner, IMO. It adds an extra check but it will use `Object.assign()`in the end, I believe.